0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+
+- `transfer.update` - subscribes to events with the exact `event` value `transfer.update` only.
+- `transfer.*` - subscribes to all events with the event value starting in `transfer.`, including at least `transfer.create` and `transfer.update`.
+- `*` - subscribes to all events. (Equivalent to omitting the `eventType` filter when subscribing.)
+
+##### Response
+The ledger acknowledges the request immediately by sending a JSON object with the following fields:
+
+| Field     | Value            | Description                                   |
+|:----------|:-----------------|:----------------------------------------------|
+| `id`      | String or Number | The `id` value from the request. This helps distinguish responses from other messages. |
+| `jsonrpc` | String           | MUST have the value `"2.0"` to indicate this is a JSON-RPC 2.0 message. |
+| `result`  | Number           | Updated number of active account subscriptions on this WebSocket connection. In practice, this is usually the length of the `params` array from the request. |
+
+Later, the ledger responds with [notifications](#websocket-notifications) whenever any of the following occurs:
+
+- A transfer affecting the account is prepared (`event` type: `transfer.create`)
+- A transfer affecting the account changes state. For example, from prepared to executed or to expired. (`event` type: `transfer.update`)
+- Someone [sends a message](#send-message) to the account. (`event` type: `message.send`)
+- A ledger MAY define additional types of notifications that get sent when you are subscribed to an account. Each new type of notification MUST have a unique `event` value.
+
+Example:
+
+```
+--> {
+      "jsonrpc": "2.0",
+      "method": "subscribe_account",
+      "params": ["https://ledger.example/accounts/alice"],
+      "id": 1
+    }
+<-- {
+      "jsonrpc": "2.0",
+      "result": 1,
+      "id": 1
+    }
+```
+
+
+#### Subscribe to Transfer
+
+The client sends a JSON object to the ledger with the following fields:
+
+| Field     | Value            | Description                                   |
+|:----------|:-----------------|:----------------------------------------------|
+| `id`      | String or Number | An arbitrary identifier for this request. MUST NOT be `null`. |
+| `jsonrpc` | String           | MUST have the value `"2.0"` to indicate this is a JSON-RPC 2.0 request. |
+| `method`  | String           | MUST be the value `"subscribe_transfer"`.      |
+| `params`  | Array            | Each member of this array must be the [URL][] of a transfer to subscribe to, as a string. This array replaces any existing transfer subscriptions on this WebSocket connection. If the array length is zero, the client is unsubscribed from all transfer notifications. |
```

Comment:
In JSONRPC 2.0, params is a "structured" value -- in other words, yes. The examples in their spec contain both.

That said, it's probably wise to make this an object so it's forward compatible and similar to the Subscribe to Account call.